### ISSUE TITLE
The right way of pre-processing for input?

### ISSUE THREAD

**Issue by godissaw** [bonlime/keras-deeplab-v3-plus#3]
![image](https://user-images.githubusercontent.com/27410324/38182818-9629429c-366e-11e8-802f-c522f09d2c74.png)
We use the first img as input, and resize it from (427, 640, 3) to (512, 512, 3), and normalize it from [0,255] to [0,1].
![image](https://user-images.githubusercontent.com/27410324/38182939-55527b16-366f-11e8-841d-9951e48c2da0.png)
As below, the output of OS=16 is normal,
![image](https://user-images.githubusercontent.com/27410324/38183044-f81ed934-366f-11e8-80a7-20660cad8b84.png)

but the output of OS=8 is worse.

I can not figure out that, so I think my pre-processiong(such as normalizing) is wrong.

**Comment by godissaw** [comment]
After trying more times with normalizing to [0,1], the OS=16 is better

**Comment by bonlime** [comment]
How to reproduce your results? I try to do this way:
`deeplab_model = Deeplabv3((512,512,3),num_classes=21, last_activation=None, OS = 16)`
`#load weights`  
`img = plt.imread("image1.jpg")`
`resized = cv2.resize(img,(512,512)) / 255.`
`res = deeplab_model.predict(np.expand_dims(resized,0))`
`plt.imshow(np.argmax(res.squeeze(),-1))`
but i get this 
![image](https://user-images.githubusercontent.com/14337581/38195527-89562464-3686-11e8-9cae-770067280533.png)

What version of TensorFlow and Keras do you use?

**Comment by godissaw** [comment]
Keras == 2.1.5
Tensorflow == 1.4.0
Cuda == 8.0, Cudnn == 6.0

**Comment by godissaw** [comment]
And I use the scipy.misc.imresize, which is different with yours

**Comment by bonlime** [comment]
Hmmm. I am using TF == 1.6.0 and Cuda==9.0.   
Maybe there are some big implementation differences between this two version

**Comment by bonlime** [comment]
@godissaw  there were huge errors in model, this is the reason why results were so bad. Fixed it. The right way of preprocessing the input is to scale it to [-1., 1.]

**Comment by godissaw** [comment]
So I just need to rescale it to[-1,1]?
thanks very much.

**Comment by ChristofHenkel** [comment]
if rescale it to[-1,1] is correct then (at least for exception backbone) this should do it:

`from keras.applications.xception import preprocess_input`
`img = preprocess_input(img)`
